src/authModel.js

```diff
@@ -0,0 +1,89 @@
+const Token = require('./models/auth/Token')
+const Client = require('./models/auth/Client')
+const AuthorizationCode = require('./models/auth/AuthorizationCode')
+
+module.exports = {
+  getAccessToken: accessToken => {
+    console.log('finding token, accessToken = ' + accessToken)
+    return Token.findOne({ accessToken })
+      .populate('user')
+  },
+
+  getAuthorizationCode: authCode => {
+    console.log('finding auth code, code = ' + authCode)
+    return AuthorizationCode.findOne({ authorizationCode: authCode })
+      .populate('user client')
+  },
+
+  getClient: (clientId, clientSecret) => {
+    console.log('finding client, id = ' + clientId + ', secret = ' + clientSecret)
+    const params = { _id: clientId }
+    if (clientSecret) params.secret = clientSecret
+    return Client.findOne(params)
+  },
+
+  getRefreshToken: refreshToken =>
+    Token.find({ refreshToken })
+      .populate('user'),
+      // .exec(),
```

Comment:
Just a heads up, I can't work out what happens when you do `.populate()` without `.exec()` because this is the problem you were having before which was only fixed by adding `.exec()`, @mattlub? I get this still passes the tests but I don't get why - have a feeling it's to do with only the client.id being the important part so maybe that can happen without `exec()` - everywhere I've looked always uses `.exec()` after `.populate()`